kube-core/src/util.rs

```diff
@@ -33,14 +33,37 @@ impl Request {
         let pparams = PatchParams::default();
         self.patch(name, &pparams, &Patch::Merge(patch))
     }
+
+    /// Cordon a resource
+    pub fn cordon(&self, name: &str) -> Result<http::Request<Vec<u8>>, request::Error> {
+        self.set_unschedulable(name, true)
+    }
+
+    /// Uncordon a resource
+    pub fn uncordon(&self, name: &str) -> Result<http::Request<Vec<u8>>, request::Error> {
+        self.set_unschedulable(name, false)
+    }
+
+    fn set_unschedulable(
+        &self,
+        node_name: &str,
+        value: bool,
+    ) -> Result<http::Request<Vec<u8>>, request::Error> {
+        self.patch(
+            node_name,
+            &PatchParams::default(),
+            &Patch::Merge(serde_json::json!({ "spec": { "unschedulable": value } })),
+        )
+    }
```

Comment:
maybe add a new `impl Request` block for these 3 functions to distinguish them from the restart helper.